### PR TITLE
Clarify dev dependencies for host mode

### PR DESCRIPTION
**Pull request by joe4dev**
We are currently missing a description of how what extra dependencies are required for running in host mode. This PR provides initial instructions and improves the dev setup guide.

As suggested by @alexrashed in a review comment [here](https://github.com/localstack/localstack/pull/10615#discussion_r1562444858).

### CODE REVIEW

File: content/en/contributing/development-environment-setup.md [hunk 1]
```diff
@@ -30,46 +30,80 @@ You will need the following tools for the local development of LocalStack.
...
 
-If you pull the repo in order to extend/modify LocalStack, run this command to install all the dependencies:
-
-{{< command >}}
-$ make install
-{{< /command >}}
+### Building the Docker image for Development
 
-This will install the required pip dependencies in a local Python `virtualenv` directory `.venv` (your global python packages will remain untouched), as well as some node modules in `./localstack/node_modules/`. Depending on your system, some `pip`/`npm` modules may require additional native libs installed.
+Please note that there are a few commands we need to run on the host to prepare the local environment for the Docker build - specifically,
```
Comment (joe4dev) [inline content/en/contributing/development-environment-setup.md:53]: @alexrashed Isn't this outdated advice with LPM?

Comment (alexrashed) [reply content/en/contributing/development-environment-setup.md:53]: Good catch! Yes, this is definitely outdated (for quite a while now 😅)!

Comment (joe4dev) [reply content/en/contributing/development-environment-setup.md]: there was even an outdated comment at the top 🙈

**Review by HarshCasper — APPROVED** [review]

File: content/en/contributing/development-environment-setup.md [hunk 2]
```diff
@@ -7,69 +7,101 @@ aliases:
...
+
+cd localstack
+source .venv/bin/activate
+
+pip install -e ../moto
+```
 
 ### Tips
 
+* Most of the contributors use the free community version of [PyCharm](https://www.jetbrains.com/pycharm/).
```
Comment (alexrashed) [inline content/en/contributing/development-environment-setup.md:104]: Well, is it necessary to state an opinion on the usage of the IDE? This shouldn't make any difference, does it?

**Review by alexrashed — COMMENTED** [review]
Thanks so much for churning over this outdated documentation! Really great to see the contribution guides being updated to simplify the onboarding of new contributors! 🚀 
I only found some small things, broken links, and some leftovers. Afterwards we're good to go! 💯 
Thanks again for jumping on this! 🦸🏽

File: content/en/contributing/development-environment-setup.md [hunk 3]
```diff
@@ -7,69 +7,101 @@ aliases:
...
-$ make install
-{{< /command >}}
+$ make docker-build
+{{< / command >}}
 
-This will install the required pip dependencies in a local Python `virtualenv` directory `.venv` (your global python packages will remain untouched), as well as some node modules in `./localstack/node_modules/`. Depending on your system, some `pip`/`npm` modules may require additional native libs installed.
+### Additional Dependencies for running LocalStack in Host Mode
 
-The Makefile contains a start command to conveniently start:
+In host mode, additional dependencies (e.g., Java) are required for developing certain AWS-emulated services (e.g., StepFunctions).
```
[the remark this answers is not in the document]
Comment (joe4dev) [reply content/en/contributing/development-environment-setup.md]: Switched to DynamoDB 👍 
(I know from almost daily reviews but didn't know an example by heart without checking. That's another reason we should make that more explicit)

File: content/en/contributing/development-environment-setup.md [hunk 4]
```diff
@@ -7,69 +7,101 @@ aliases:
...
+The required dependencies vary depending on the service, [Configuration](https://docs.localstack.cloud/references/configuration/), operating system, and system architecture (i.e., x86 vs ARM).
```
[the remark this answers is not in the document]
Comment (joe4dev) [reply content/en/contributing/development-environment-setup.md]: That was an attempt to anticipate the move to Github (see discussion in Notion ;) but I'm happy to change it back for now.

IHMO, these instructions co-evolve with the code and only get updated if we move it into GitHub.

File: content/en/contributing/development-environment-setup.md [hunk 5]
```diff
@@ -7,69 +7,101 @@ aliases:
...
+Refer to our official [Dockerfile](https://github.com/localstack/localstack/blob/master/Dockerfile) and our [package installer LPM](./Concepts/index.md#packages-and-installers) for more details.
```
[the remark this answers is not in the document]
Comment (joe4dev) [reply content/en/contributing/development-environment-setup.md]: fixed by also converting it into a Hugo ref

File: content/en/contributing/development-environment-setup.md [hunk 6]
```diff
@@ -7,69 +7,101 @@ aliases:
...
-{{< / command >}}
+#### Python Dependencies
 
-### Building the Docker image for Development
+* [JPype1](https://pypi.org/project/JPype1/) might require `g++` to fix a compile error on ARM Linux `gcc: fatal error: cannot execute ‘cc1plus’`
+  * Used in EventBridge, EventBridge Pipes, and Lambda Event Source Mapping for a Java-based event ruler via the opt-in configuration `EVENT_RULE_ENGINE=java`
+  * Introduced in [#10615](https://github.com/localstack/localstack/pull/10615)
+* [libvirt-python](https://pypi.org/project/libvirt-python/) requires `libvirt-dev` on Debian or `libvirt` on macOS/Brew to fix `Package libvirt was not found in the pkg-config search path.`
+  * Used in EC2 to access Libvirt inside the LocaStack container
+  * Introduced in [localstack-ext#2827](https://github.com/localstack/localstack-ext/pull/2827)
```
[the remark this answers is not in the document]
Comment (joe4dev) [reply content/en/contributing/development-environment-setup.md]: done.

another reason to have it in GitHub and add it to localstack-ext as well ;)

File: content/en/contributing/development-environment-setup.md [hunk 7]
```diff
@@ -7,69 +7,101 @@ aliases:
...
 
-We generally recommend using this command to build the Docker image locally (works on Linux/MacOS):
+Some services or tests might need some of these dependencies (not yet categorized):
 
-{{< command >}}
-$ make docker-build
-{{< / command >}}
+* [OpenJDK](https://openjdk.org/install/)
+* [NodeJS & npm](https://nodejs.org/en/download/)
+* [Maven](https://maven.apache.org/download.cgi)
```
[the remark this answers is not in the document]
Comment (joe4dev) [reply content/en/contributing/development-environment-setup.md]: removed (I guess Maven & Gradle were used in some tests for packaging but these should be migrated to Docker-based builds now)

File: content/en/contributing/development-environment-setup.md (hunk 2 above)
Comment (joe4dev) [reply content/en/contributing/development-environment-setup.md]: removed
(it was my attempt to hide it here, given it was the prominent step 3. of the setup instructions before 😄 )

File: content/en/contributing/development-environment-setup.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (alexrashed) [reply content/en/contributing/development-environment-setup.md]: Sure, but that's a different debate. Let's stick to the best practices for the current platform, and migrate everything properly when moving parts to another platform. 😉

File: content/en/contributing/development-environment-setup.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (alexrashed) [reply content/en/contributing/development-environment-setup.md]: Actually they were used in some java-based services to download additional dependencies iirc.

**Review by alexrashed — APPROVED** [review]
Awesome! Thanks for addressing all the comments! Lookin' great! 🚀

**Comment by joe4dev** [comment]
@bentsku I re-added Node.js as a mandatory dependency given that it is required to run any test. Thanks for your input 👍

**Pull request merged**